src/main/java/com/fakng/fakngagrgtr/controller/CustomControllerAdvice.java

```diff
@@ -0,0 +1,73 @@
+package com.fakng.fakngagrgtr.controller;
+
+import com.fakng.fakngagrgtr.enumiration.FakngError;
+import com.fakng.fakngagrgtr.exception.FakngException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+import javax.xml.bind.ValidationException;
+
+/**
+ * Централизированное управление ошибками.<br>
+ * Все ошибки летят сюда.
+ *
+ * В любой момент можем добавить другой метод, который ловит определенный тип ошибок
+ */
+@Slf4j
+@RestControllerAdvice
+class CustomControllerAdvice {
+
+    /**
+     * Если летит наша ошибка
+     * @param e выброшенный exception
+     * @return ответ с ошибкой 400
+     */
+    @ExceptionHandler(FakngException.class)
+    public ResponseEntity<String> handleLibraryExceptions(FakngException e) {
+        log.error(e.getClass().toString());
+        log.error(e.getMessage());
+        for (StackTraceElement element : e.getStackTrace())
```

Comment:
You shouldn't do this. Logger usually is able to log the entire stacktrace. There's no need to loop through it.

Reply:
OK, but are we adding stack trace to the log or not?